EWL-6955 Sticky Main Navigation

designerneil opened the pull request:
## Ticket(s)

**Jira Ticket**
- [EWL-6955: Sticky Nav scrolling for AMA One](https://issues.ama-assn.org/browse/EWL-6955)

## Description
As a user on the site ama-assn.org. I would like the ability to scroll down the page and have the purple navigation block/bar remain at the top so that I can easily see and click the links without having to scroll back up to the top. 

## To Test
- [x] switch branch to `feature/EWL-6955-sticky-nav`
- [x] run `gulp serve`
- [x] navigate to pages > homepage
- [x] scroll page
- [x] observe how the the main purple navigation stays in place
- [x] resize screen to mobile and tablet
- [x] observe how the the main purple navigation stays in place
- [x] Did you test in IE 11?

## Visual Regressions
A report is available [here](http://htmlpreview.github.io/?https://github.com/AmericanMedicalAssociation/ama-style-guide-2/blob/visual-regression-testing-artifact/feature/EWL-6955-sticky-nav/html_report/index.html).


## Relevant Screenshots/GIFs
![screen shot 2019-03-05 at 3 58 33 pm](https://user-images.githubusercontent.com/2271747/53840376-907c0280-3f5f-11e9-940e-27c12e3f8a60.png)


## Remaining Tasks
Test in D8 


## Additional Notes
This PR is related to a PR in D8 `feature/EWL-6955-sticky-nav`

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

designerneil committed:
Adds sticky library and updates main nav to use it

Uses sticky library from https://github.com/garand/sticky. Uses method to make main nav sticky when a user scroll

designerneil committed:
Accounts top spacing for main nav for when user is logged in

designerneil committed:
Implements sticky social icons

Leverages the sticks library to make the social icons sticky

designerneil committed:
Updates article class to be compatible with SG2 articles

designerneil committed:
removes sticky social icons

This was added by accident

designerneil committed:
Updates main navigation to scroll when it doesn’t fit inside the viewport

JaeBea commented:
This must be tested in AMA One. Adding flag.

JaeBea commented on styleguide/build.config.json:
```diff
@@ -32,6 +32,7 @@
       "source/assets/js/vendor/jquery-ui.accordion.multiple.js",
       "source/assets/js/vendor/jquery.smartmenus.js",
+      "source/assets/js/vendor/jquery.sticky.js",
       "source/assets/js/**/*.js"
```
@designerneil ,
The sticky script file needs to be added to drupal files section of this file

JaeBea reviewed (changes requested):
@designerneil ,
Two changes are requested.
1. There is a jump that happens when you load a page and start to scroll on Chrome. I recorded a video to show you what I am experiencing. It has been uploaded to drive and [can be viewed here](https://drive.google.com/file/d/1s-doLZqyM75MOx3fYqJXvSzhrQy9wPAX/view?usp=sharing). Could you prevent that jump?

2. Described further below.

designerneil commented:
@JaeBea I didn't want mix vendor stuff with our stuff. The current pattern is not to mix the two. For example the fitvid.js is load separately. 

As for the page jumping. If you load the page without the iframe it no longer jumps. http://localhost:3000/patterns/05-pages-resource/05-pages-resource.html

JaeBea commented:
@designerneil ,
I am getting the jumping in D8 as an anonymous user as well. I have attempted loading the SG2 page outside of the iframe and get the same jump.

designerneil committed:
Adds condition to check if body has overflow hidden

If body has overflow hidden then don’t make the main nav sticky

Adds condition to check if body has overflow hidden

designerneil committed:
Adds condition to check if body has overflow hidden

If body has overflow hidden then don’t make the main nav sticky

designerneil committed:
Adds condition to check if body has overflow hidden

designerneil committed:
Merge branch 'feature/EWL-6955-sticky-nav' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-6955-sticky-nav

designerneil committed:
Merge branch 'develop' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-6955-sticky-nav

designerneil commented:
@JaeBea the stickynav jumping issue has been addressed. Thanks

JaeBea commented:
@designerneil ,
I am still experiencing the jumping in d8. I am going to request a second review to confirm this is not an issue only with me.

mgrahamama reviewed (approved):
The stickynav functions smooth on chrome, safari, ff, and ie11

designerneil committed:
Updates javascript to factor in viewport weirdness affecting submenu scrolling

designerneil committed:
Calculate visible height of article instead of window

designerneil committed:
Clarifies $productNavHeight variable

designerneil committed:
update scrollBot to scrollBottom

tcfunk reviewed (changes requested):
This seems to work very well on both desktop and tablet, great job 👍 

On mobile (especially on an actual device), the stickiness works just fine, but I am experiencing just a couple issues:
- Due to the stickiness, I can't access the full nav. I can see everything down to "AMA Ed Hub" under "Explore Products" and the rest is cut off. Also if I expand the About section I can't see all the items under it. (Device was iPhone 6+)
- ~This one might not be related to this PR, but after I expand a section in the navigation, if I try to close that section again I'm actually taken to the primary category page instead. E.g., if I expand About, then try to close that section, I end up on the About page.~
- ~Last thing, and we might not care about this, and it is likely unrelated to this PR as well, but the homepage looks pretty weird at mobile resolutions on IE 11 https://puu.sh/DketI/773e1b6392.png~

I'm going to try out the second and third issues that I mentioned on the develop branch, and if they persist then I'll udpate my comment here and make note of those issues elsewhere.

Edit: crossed out issues that are present on develop

designerneil committed:
Fixes menu position

designerneil committed:
Removes previous commit

designerneil committed:
Adds layout-container and updates js accordingly

designerneil committed:
Removes commented code

designerneil commented:
@tcfunk `Due to the stickiness, I can't access the full nav. I can see everything down to "AMA Ed Hub" under "Explore Products" and the rest is cut off. Also if I expand the About section I can't see all the items under it. (Device was iPhone 6+)
`
Should be taken care of in PR  https://github.com/AmericanMedicalAssociation/ama-style-guide-2/pull/601

designerneil committed:
Merge branch 'develop' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-6955-sticky-social-icons

designerneil committed:
Merge branch 'feature/EWL-6638-menu-expansion' of github.com:AmericanMedicalAssociation/ama-style-guide-2 into feature/EWL-6955-sticky-social-icons

JaeBea reviewed (approved):
I seem to be the only one with the jumping issue. I will assume it is something with my computer and approve.
